### PR TITLE
WIP mark archives generated by gym with extended attribute

### PR DESCRIPTION
**Pull request by mpirri**
NOTE This PR is on my own fork of gym because I do not wish to have discussion of this in the public fastlane repos (since this change is Fabric-internal motivated).  Hemal agreed with the decision to do it this way.

This is a change to mark archives generated by gym to allow them to be differentiated from archives generated by Xcode. This ability is required by the Mac app (changes are in a separate PR) to allow it to disable the "distribute this archive with Crashlytics Beta" user notification & prompt, which currently fires after gym finishes building any archive, and is confusing for users.

Biggest concern with this change comes with the fact that we are adding functionality to fastlane/gym to support Fabric mac app. Although it is not impossible to imagine a situation where this might be useful to another tool, it is most likely to only ever be used by Fabric.

How do we feel about this? Modifying our open-source project with code to support closed-source Fabric functionality? Thoughts?

### CODE REVIEW

**Comment by ohwutup** [comment]
This seems pretty chill to me. I think that it could definitely be useful to have this flag for other purposes beyond ours. :rocket:

File: lib/gym/runner.rb [hunk 1]
```diff
@@ -71,6 +71,10 @@ def fix_package
       Gym::XcodebuildFixes.watchkit2_fix
     end
 
+    def mark_archive_as_built_by_gym(archive_path)
+      system "xattr -w info.fastlane.generated_by_gym 1 \"#{archive_path}\""
```
Comment (KrauseFx) [inline lib/gym/runner.rb]: Please surround with `(`

Comment (KrauseFx) [inline lib/gym/runner.rb]: The `"` archive_path will have issues if there is a `"` in the file name. Maybe we should use the `shellescape` method instead.

**Comment by KrauseFx** [comment]
Great choice on doing a private fork on this - great idea using the `xcattr` method to do so. The only thing that we need to check first, is that this change doesn't conflict with the iTunes Connect upload and app submission. In particular I wonder if this change somehow breaks the code signing attributes of the binary. Probably not, as it only affects the archive, but we still need to test this :+1:

**Comment by KrauseFx** [comment]
Uploading a binary to iTunes Connect and see if it can be processed should be enough to test this.

**Comment by mpirri** [comment]
@KrauseFx I checked uploading to iTunes Connect and then addressed your PR feedback and opened a PR on the fastlane/gym repository: https://github.com/fastlane/gym/pull/206
Let me know what you think!

@ohwutup here is the associated Mac app change to look at the xattr and the disable the beta prompt if it is set: https://github.com/crashlytics/apple/pull/4322

**Pull request closed without merging**